core/store/migrations/migration1585559482/migrate.go

```diff
@@ -0,0 +1,51 @@
+package migration1585559482
+
+import (
+	"github.com/jinzhu/gorm"
+)
+
+// Split pending_confirmations into pending_outgoing_confirmations and pending_incoming_confirmations
+// For both task_runs and job_runs
+// This is inherently ambiguous (that's the entire reason for splitting them out) but we can make a best
+// guess by checking for presence of a transaction.
+func Migrate(tx *gorm.DB) error {
+	// HACK: Assume it's pending outgoing transactions if there exists a TX for the job run owning this task run
+	err := tx.Exec(`
+		UPDATE task_runs
```

Comment:
On some of our nodes we have many millions of TaskRuns, might want to double check this doesn't take too long as it will cause downtime.

Reply:
On our prod database, this takes under 100ms to run all statements.

However, I will address your other comment and perhaps a migration may not even be necessary.